src/windows.rs

```diff
@@ -0,0 +1,464 @@
+                let mut info: MONITORINFOEXW = std::mem::MaybeUninit::zeroed().assume_init();
+                info.__AnonymousBase_winuser_L13558_C43.cbSize =
+                    std::mem::size_of::<MONITORINFOEXW>() as u32;
+                let info_ptr = &mut info as *mut _;
+                if !GetMonitorInfoW(hmonitor, info_ptr as *mut MONITORINFO).as_bool() {
+                    return vec![Err(SysError::ListingDevicesFailed(
+                        "GetMonitorInfoW",
+                        GetLastError(),
+                    ))];
+                };
+
+                // Get the Physical Monitors in the HMONITOR
+                let mut physical_number: u32 = 0;
+                if !BOOL(GetNumberOfPhysicalMonitorsFromHMONITOR(
+                    hmonitor,
+                    &mut physical_number,
+                ))
+                .as_bool()
+                {
+                    return vec![Err(SysError::ListingDevicesFailed(
+                        "GetNumberOfPhysicalMonitorsFromHMONITOR",
+                        GetLastError(),
+                    ))];
+                };
+                let mut physical_monitors = Vec::with_capacity(physical_number as usize);
+                if !BOOL(GetPhysicalMonitorsFromHMONITOR(
+                    &hmonitor,
+                    physical_number,
+                    physical_monitors.as_mut_ptr(),
+                ))
+                .as_bool()
+                {
+                    return vec![Err(SysError::ListingDevicesFailed(
+                        "GetPhysicalMonitorsFromHMONITOR",
+                        GetLastError(),
+                    ))];
+                }
+                physical_monitors.set_len(physical_number as usize);
+
+                // Get the Display Devices in the HMONITOR
+                let mut device_number = 0;
+                let mut device: DISPLAY_DEVICEW = std::mem::MaybeUninit::zeroed().assume_init();
+                device.cb = std::mem::size_of::<DISPLAY_DEVICEW>() as u32;
+                let mut display_devices = Vec::new();
+                while EnumDisplayDevicesW(
+                    PWSTR(info.szDevice.as_mut_ptr()),
+                    device_number,
+                    &mut device,
+                    EDD_GET_DEVICE_INTERFACE_NAME,
+                )
+                .as_bool()
+                {
+                    device_number = device_number + 1;
+                    display_devices.push(device.clone());
+                }
+                if display_devices.len() != physical_monitors.len() {
+                    // There doesn't seem to be anyway to directly associate a physical monitor handle
+                    // with the equivalent display device, other than by array indexing
+                    // https://stackoverflow.com/questions/63095216/how-to-associate-physical-monitor-with-monitor-deviceid
+                    return vec![Err(SysError::EnumerationMismatch)];
```

Comment:
This early-return leaks the physical monitor handles. They need to be wrapped in a type implementing `Drop` immediately after receiving them in order to avoid leaking them in all scenarios.